url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/iomodel/Player.java

```diff
@@ -0,0 +1,1017 @@
+	 */
+	public int getBlockZ() {
+		return Z / 32;
+	}
+	/**
+	 * Set the current X pos of the player
+	 * This will NOT teleport the player
+	 * To teleport the player, use {@link #setPos(short, short, short)}
+	 * @param value
+	 */
+	public void setX(short value) {
+		oldX = X;
+		X = value;
+	}
+	/**
+	 * Set the current Y pos of the player
+	 * This will NOT teleport the player
+	 * To teleport the player, use {@link #setPos(short, short, short)}
+	 * @param value
+	 */
+	public void setY(short value) {
+		oldY = Y;
+		Y = value;
+	}
+	/**
+	 * Set the current Z pos of the player
+	 * This will NOT teleport the player
+	 * To teleport the player, use {@link #setPos(short, short, short)}
+	 * @param value
+	 */
+	public void setZ(short value) {
+		oldZ = Z;
+		Z = value;
+	}
+	/**
+	 * Teleport the player
+	 * @param x The X pos to teleport to
+	 * @param y The Y pos to teleport to
+	 * @param z The Z pos to teleport to
+	 * @param yaw The new yaw
+	 * @param pitch The new pitch
+	 */
+	public void setPos(short x, short y, short z, byte yaw, byte pitch) {
+		setX(x);
+		setY(y);
+		setZ(z);
+		oldyaw = this.yaw;
+		this.yaw = yaw;
+		oldpitch = this.pitch;
+		this.pitch = pitch;
+		TP();
+	}
+	/**
+	 * Teleport the player
+	 * @param x The X pos to teleport to
+	 * @param y The Y pos to teleport to
+	 * @param z The Z pos to teleport to
+	 */
+	public void setPos(short x, short y, short z) {
+		setPos(x, y, z, yaw, pitch);
+	}
+	
+	protected void TP() {
+		Packet p = pm.getPacket("TP");
+		for (Player pp : pm.server.players) {
+			if (pp.level == level)
+				p.Write(pp, pp.pm.server, this, ID); //Tell all the other players as well...
+		}
+	}
+	
+	/**
+	 * Sends a message to the player.
+	 * If the message is longer than 64 chars, then it will not send.
+	 * 
+	 * @param message
+	 *               The message to send
+	 */
+	@Override
+	public void sendMessage(String message){
+		Packet p = pm.getPacket("Message");
+		String[] messages = chat.split(message);
+		for (String m : messages) {
+			this.message = m;
+			p.Write(this, pm.server);
+		}
+	}
+	/**
+	 * Change the level the player is currently in. This method will
+	 * call {@link #changeLevel(Level, boolean)} with threading being false.
+	 * So this method will block until the level sending is finished.
+	 * @param level The new level the player will be moved to.
+	 */
+	public void changeLevel(Level level) {
+		changeLevel(level, false);
+	}
+	/**
+	 * Change the level the player is currently in. This method will
+	 * block if threading is false. If threading is true, level sending will
+	 * begin in a separate thread and this method wont block.
+	 * @param level The new level the player will be moved to.
+	 * @param threading Weather to make this call in a separate thread or not.
+	 */
+	public void changeLevel(Level level, boolean threading)
+	{
+		if (!threading) {
+			setLevel(level);
+			try {
+				levelsender.join(); //Wait for finish
+			} catch (InterruptedException e) { } 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)